app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/interactor/SubscriberDetailInteractorImpl.java

```diff
@@ -41,8 +46,13 @@ public void getUserRepositoryData(String userName, final OnFinishedListener list
         call.enqueue(new Callback<List<Repository>>() {
             @Override
             public void onResponse(Call<List<Repository>> call, Response<List<Repository>> response) {
-                List<Repository> repositoryList = response.body();
-                listener.onFinishedRepository(repositoryList);
+                if (response.isSuccessful()) {
+                    List<Repository> repositoryList = response.body();
+                    listener.onFinishedRepository(repositoryList);
+                } else {
+                    listener.onFailureRepository(Constants.MESSAGE_FAILED_SERVICE);
```

Comment:
The flow is, Interactor passes an error and type to the presenter, the presenter calls show error message on the view, the view shows the proper string.